directives/questionnaire.py

```diff
@@ -108,6 +109,7 @@ def run(self):
                 u'fields': (u'#!children', None),
             }],
         }
+        data.setdefault('status', self.options.get('status', 'unlisted'))    
```

Comment:
In this case, `setdefault` always sets the key since the key `status` has never been set to the dict before this line. Thus, you could just set the key without calling setdefault. You can do it in the same statement that sets all those other keys.

Reply:
In the submit directive, the data dict may be read from a YAML file so the status key could be already there. That is why setdefault made sense in the submit directive.